### PR TITLE
CORTX-31656 Enable ADDB plugin generation during custom ci-job

### PR DESCRIPTION
**Pull request by Abhirup-A-G**
Signed-off-by: Abhirup Ghosh <abhirup.ghosh@seagate.com>

# Problem Statement
- Requester wanted to add input parameter for the custom-ci job [https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/build?delay=0sec) ] to enable/disable ADDB plugin generation.

Modification required in the below scripts:

https://github.com/Abhirup-A-G/cortx-re/blob/866859cc64c26f47f7cdf20be9a3999eb5f8918a/jenkins/custom-ci/generic/cortx-rgw-integration.groovy#L43
https://github.com/Seagate/cortx-re/blob/8f2bc4386b0fccce4a01f71c2f71799dc6a8840c/jenkins/custom-ci/generic/custom-ci.groovy#L183

# Design
-  Bug: Its still failing due to building rpm package.
[Release_Engineering » re-workspace » AbhirupSpace » cortx-rgw-integration #3 Console [Jenkins] (seagate.com)](https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/AbhirupSpace/job/cortx-rgw-integration/3/console)

-  Jenkins Job: https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/AbhirupSpace/job/ADDB%20plugin%20generation/9/

- Updated jenkins Job ran from the developer branch [ https://github.com/Seagate/cortx-rgw-integration/pull/99 ] https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/AbhirupSpace/job/ADDB%20plugin%20generation/10/
- Jenkins job ran from the main branch : https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/AbhirupSpace/job/ADDB%20plugin%20generation/11/

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/custom-ci/generic/cortx-rgw-integration.groovy [hunk 1]
```diff
@@ -11,6 +11,12 @@ pipeline {
         string(name: 'CORTX_RGW_INTEGRATION_BRANCH', defaultValue: 'main', description: 'Branch for cortx-rgw-integration.')
         string(name: 'CUSTOM_CI_BUILD_ID', defaultValue: '0', description: 'Custom CI Build Number')
         // Add os_version parameter in jenkins configuration
+
+        choice(
+            name: 'ENABLE_ADDB_PLUGIN',
+                choices: ['yes', 'no'],
```
Comment (nitisdev) [inline jenkins/custom-ci/generic/cortx-rgw-integration.groovy:17]: Please put default choice as "no". Developers can select when they need to build addb plugin.

Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy:17]: Done

Comment (gauravchaudhari02) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy:17]: @Abhirup-A-G, get it clarified with developers if they want default as `yes` or `no` each time.

File: jenkins/custom-ci/generic/custom-ci.groovy [hunk 2]
```diff
@@ -185,11 +191,12 @@ pipeline {
                         script { build_stage = env.STAGE_NAME }
                         script {
                             try {
-                                def rgwintegrationbuild = build job: '/GitHub-custom-ci-builds/generic/cortx-rgw-integration-build', wait: true,
+                                def rgwintegrationbuild = build job: '/Release_Engineering/re-workspace/AbhirupSpace/cortx-rgw-integration', wait: true,
```
Comment (gauravchaudhari02) [inline jenkins/custom-ci/generic/custom-ci.groovy:194]: change job to original integration build instead of your test job.

Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/custom-ci.groovy:194]: After testing i will revert back to original. Its for testing .

**Comment by gauravchaudhari02** [comment]
@Abhirup-A-G, please add tested job links in test section

File: jenkins/custom-ci/generic/cortx-rgw-integration.groovy [hunk 3]
```diff
@@ -41,7 +47,15 @@ pipeline {
                 script { build_stage = env.STAGE_NAME }
 
                 sh encoding: 'UTF-8', label: 'cortx-provisioner', script: '''
-                bash ./jenkins/build.sh -v 2.0.0 -b ${CUSTOM_CI_BUILD_ID}
+                    if [ "${ENABLE_ADDB_PLUGIN}" == "yes" ]; then
+                        pushd ../
```
Comment (Abhirup-A-G) [inline jenkins/custom-ci/generic/cortx-rgw-integration.groovy:51]: I will Add the input parameter for the branch name and branch URL and export them here post successful testing.

File: jenkins/custom-ci/generic/cortx-rgw-integration.groovy [hunk 4]
```diff
@@ -41,7 +47,15 @@ pipeline {
...
+                        git clone https://github.com/Seagate/cortx-motr.git && git clone https://github.com/Seagate/cortx-rgw
+                        popd
+                        bash ./jenkins/build.sh -v 2.0.0 -b ${CUSTOM_CI_BUILD_ID} -addb
+                    else
```
Comment (Abhirup-A-G) [inline jenkins/custom-ci/generic/cortx-rgw-integration.groovy:55]: I will Add the input parameter for the branch name and branch URL and export them here post successful testing.

File: jenkins/custom-ci/generic/custom-ci.groovy (hunk 2 above)
Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/custom-ci.groovy:194]: Done

File: jenkins/custom-ci/generic/cortx-rgw-integration.groovy (hunk 4 above)
Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy:55]: After discussing with @shailesh-vaidya , I am keeping the Git clone logic in the code

File: jenkins/custom-ci/generic/custom-ci.groovy [hunk 5]
```diff
@@ -80,6 +80,12 @@ pipeline {
                 description: 'Build motr rpm using dtm mode.'
         )
 
+        choice(
+            name: 'ENABLE_ADDB_PLUGIN',
+                choices: ['no', 'yes'],
```
Comment (shailesh-vaidya) [inline jenkins/custom-ci/generic/custom-ci.groovy:85]: Default ENABLE_ADDB_PLUGIN is `yes` at https://github.com/Abhirup-A-G/cortx-re/blob/CORTX-31656-Updated/jenkins/custom-ci/generic/cortx-rgw-integration.groovy#L17 
We should do the same here also.

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: jenkins/custom-ci/generic/custom-ci.groovy (hunk 5 above)
Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/custom-ci.groovy:85]: Default needs to be no as requested by user. I have updated the code accordingly.

File: jenkins/custom-ci/generic/cortx-rgw-integration.groovy (hunk 1 above)
Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy]: They wanted it as No

File: jenkins/custom-ci/generic/cortx-rgw-integration.groovy (hunk 4 above)
Comment (shailesh-vaidya) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy:55]: > I will Add the input parameter for the branch name and branch URL and export them here post successful testing.

@Abhirup-A-G Please address this. Use repo_url and branch name provided for custom-ci input.

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: jenkins/custom-ci/generic/cortx-rgw-integration.groovy (hunk 4 above)
Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy:55]: Done

File: jenkins/custom-ci/generic/cortx-rgw-integration.groovy [hunk 6]
```diff
@@ -40,8 +50,16 @@ pipeline {
             steps {
                 script { build_stage = env.STAGE_NAME }
 
-                sh encoding: 'UTF-8', label: 'cortx-provisioner', script: '''
-                bash ./jenkins/build.sh -v 2.0.0 -b ${CUSTOM_CI_BUILD_ID}
+                sh encoding: 'UTF-8', label: 'cortx-provisioner', script: '''          
+                    if [ "${ENABLE_ADDB_PLUGIN}" == "yes" ]; then
+                        pushd ../
+                        git clone $MOTR_URL && git clone $CORTX_RGW_URL
```
[the remark this answers is not in the document]
Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy]: Done.

[the remark this answers is not in the document]
Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy]: sh encoding: 'UTF-8', label: 'cortx-provisioner', script: '''          
                    if [ "${ENABLE_ADDB_PLUGIN}" == "yes" ]; then
                        pushd ../
                        git clone -b $MOTR_BRANCH $MOTR_URL && git clone -b $CORTX_RGW_BRANCH $CORTX_RGW_URL
                        popd
                        bash ./jenkins/build.sh -v 2.0.0 -b ${CUSTOM_CI_BUILD_ID} -addb
                    else
                        bash ./jenkins/build.sh -v 2.0.0 -b ${CUSTOM_CI_BUILD_ID}
                    fi
    
                '''

[the remark this answers is not in the document]
Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy]: `sh encoding: 'UTF-8', label: 'cortx-provisioner', script: '''          
                    if [ "${ENABLE_ADDB_PLUGIN}" == "yes" ]; then
                        pushd ../
                        git clone -b $MOTR_BRANCH $MOTR_URL && git clone -b $CORTX_RGW_BRANCH $CORTX_RGW_URL
                        popd
                        bash ./jenkins/build.sh -v 2.0.0 -b ${CUSTOM_CI_BUILD_ID} -addb
                    else
                        bash ./jenkins/build.sh -v 2.0.0 -b ${CUSTOM_CI_BUILD_ID}
                    fi
    
                '''`

[the remark this answers is not in the document]
Comment (Abhirup-A-G) [reply jenkins/custom-ci/generic/cortx-rgw-integration.groovy]: sh encoding: 'UTF-8', label: 'cortx-provisioner', script: '''          
                    if [ "${ENABLE_ADDB_PLUGIN}" == "yes" ]; then
                        pushd ../
                        git clone -b $MOTR_BRANCH $MOTR_URL && git clone -b $CORTX_RGW_BRANCH $CORTX_RGW_URL
                        popd
                        bash ./jenkins/build.sh -v 2.0.0 -b ${CUSTOM_CI_BUILD_ID} -addb
                    else
                        bash ./jenkins/build.sh -v 2.0.0 -b ${CUSTOM_CI_BUILD_ID}
                    fi
                '''

**Review by shailesh-vaidya — APPROVED** [review]

**Review by nitisdev — APPROVED** [review]

**Pull request merged**